src/components/header.js

```diff
@@ -1,6 +1,5 @@
 // Node modules import
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom';
```

Comment:
Everything was correct except this line. 
It was just imported from the wrong library...
Instead, should be:
import { Link } from 'react-router';